independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorGenerator.java

```diff
@@ -224,38 +228,72 @@ private void addIntercept(MethodCreator intercept, MethodInfo interceptorMethod,
                 retType = Object.class;
             } else {
                 // @PostConstruct, @PreDestroy, @AroundConstruct
-                retType = interceptorMethod.returnType().kind().equals(Type.Kind.VOID) ? void.class : Object.class;
+                retType = interceptorMethods.get(0).returnType().kind().equals(Type.Kind.VOID) ? void.class : Object.class;
             }
-            ResultHandle ret;
-            // Check if interceptor method uses InvocationContext or ArcInvocationContext
-            Class<?> invocationContextClass;
-            if (interceptorMethod.parameters().get(0).name().equals(DotNames.INVOCATION_CONTEXT)) {
-                invocationContextClass = InvocationContext.class;
-            } else {
-                invocationContextClass = ArcInvocationContext.class;
+
+            int i = 0;
```

Comment:
This index is never increased, or?

Reply:
L290?

Reply:
You're right. I must have missed that somehow :roll_eyes: